### PR TITLE
MessageParameterAttribute made public

### PR DESCRIPTION
**Pull request by g7ed6e**
#345

### CODE REVIEW

File: src/CoreWCF.Primitives/src/CoreWCF/Description/CustomAttributeProvider.cs [hunk 1]
```diff
@@ -246,15 +254,28 @@ private static MessageBodyMemberAttribute ConvertFromServiceModelMessageBodyMemb
...
+
+        private static MessageParameterAttribute ConvertFromServiceModelMessageParameterAttribute(object attr)
+        {
+            var messageParameter = new MessageParameterAttribute();
+            string tmpStr = GetProperty<string>(attr, nameof(MessageParameterAttribute.Name));
+            if (!string.IsNullOrEmpty(tmpStr))
+            {
+                messageParameter.Name = tmpStr;
+            }
+
```
Comment (birojnayak) [inline src/CoreWCF.Primitives/src/CoreWCF/Description/CustomAttributeProvider.cs:275]: --nit-- empty line

File: src/CoreWCF.Primitives/src/CoreWCF/MessageParameterAttribute.cs [hunk 2]
```diff
@@ -6,7 +6,7 @@
 namespace CoreWCF
 {
     [AttributeUsage(CoreWCFAttributeTargets.Parameter, Inherited = false)]
-    internal sealed class MessageParameterAttribute : Attribute
+    public sealed class MessageParameterAttribute : Attribute
```
Comment (birojnayak) [inline src/CoreWCF.Primitives/src/CoreWCF/MessageParameterAttribute.cs:9]: this should be public as per https://referencesource.microsoft.com/#System.ServiceModel/System/ServiceModel/MessageParameterAttribute.cs,98a743085573662e ... @mconnew do you have any concern ?

File: src/CoreWCF.Primitives/src/CoreWCF/Description/CustomAttributeProvider.cs [hunk 3]
```diff
@@ -116,7 +116,7 @@ public object[] GetCustomAttributes(Type attributeType, bool inherit)
                     result = attributes.Where(attribute => attribute.GetType().FullName.Equals(ServiceReflector.SMMessageHeaderAttributeFullName)).ToArray();
                     for (int i = 0; i < result.Length; i++)
                     {
-                        result[i] = ConvertFromServiceModelMessageHeadertAttribute(result[i]);
+                        result[i] = ConvertFromServiceModelMessageHeaderAttribute(result[i]);
```
Comment (birojnayak) [inline src/CoreWCF.Primitives/src/CoreWCF/Description/CustomAttributeProvider.cs:119]: 👍

File: src/CoreWCF.Primitives/src/CoreWCF/Description/CustomAttributeProvider.cs (hunk 1 above)
Comment (mconnew) [reply src/CoreWCF.Primitives/src/CoreWCF/Description/CustomAttributeProvider.cs:275]: There should be an empty line after a closing brace of an if block

File: src/CoreWCF.Primitives/src/CoreWCF/MessageParameterAttribute.cs (hunk 2 above)
Comment (mconnew) [reply src/CoreWCF.Primitives/src/CoreWCF/MessageParameterAttribute.cs:9]: No concern, this is the whole point of this pr 😃.

**Comment by mconnew** [comment]
Thankyou @g7ed6e

**Pull request merged**